### ISSUE TITLE
Test failure in object_table_tests.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#359]
Occasionally, the object table tests in 

```
cd python/ray/core
bash ../../../src/common/test/run_tests.sh
```

Fail with

```
* Suite object_table_tests:
..[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_lookup, retry_count = 5
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_lookup, retry_count = 4
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_lookup, retry_count = 3
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_lookup, retry_count = 2
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_lookup, retry_count = 1
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_lookup, retry_count = 0
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:91) Table command object_table_lookup with timer ID 0 failed
.[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_add, retry_count = 5
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_add, retry_count = 4
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_add, retry_count = 3
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_add, retry_count = 2
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_add, retry_count = 1
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_add, retry_count = 0
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:91) Table command object_table_add with timer ID 0 failed
.[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_subscribe_to_notifications, retry_count = 5
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_subscribe_to_notifications, retry_count = 4
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_subscribe_to_notifications, retry_count = 3
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_subscribe_to_notifications, retry_count = 2
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_subscribe_to_notifications, retry_count = 1
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_subscribe_to_notifications, retry_count = 0
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:91) Table command object_table_subscribe_to_notifications with timer ID 0 failed
...[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_lookup, retry_count = 0
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:91) Table command object_table_lookup with timer ID 0 failed
.[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_add, retry_count = 0
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:91) Table command object_table_add with timer ID 0 failed
.[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_subscribe_to_notifications, retry_count = 0
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:91) Table command object_table_subscribe_to_notifications with timer ID 0 failed
.[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:85) retrying operation object_table_subscribe_to_notifications, retry_count = 0
[WARN] (/Users/travis/build/ray-project/ray/src/common/state/table.cc:91) Table command object_table_subscribe_to_notifications with timer ID 0 failed
[FATAL] (/Users/travis/build/ray-project/ray/src/common/test/object_table_tests.cc:581: errno: Operation now in progress) Check failure: 0 
0   object_table_tests                  0x000000010f548df0 _Z31subscribe_success_fail_callback8UniqueIDPvS0_ + 112
1   object_table_tests                  0x000000010f5576d0 _Z21table_timeout_handlerP11aeEventLoopxPv + 256
2   object_table_tests                  0x000000010f5595e5 aeProcessEvents + 1077
3   object_table_tests                  0x000000010f55975b aeMain + 43
4   object_table_tests                  0x000000010f54ac7a _Z18object_table_testsv + 6218
5   object_table_tests                  0x000000010f54c3f4 main + 788
6   libdyld.dylib                       0x00007fff8bbfc5ad start + 1
../../../src/common/test/run_tests.sh: line 16:  8346 Abort trap: 6           ./src/common/object_table_tests
```

For example, this occurs here https://travis-ci.org/ray-project/ray/jobs/210109127.

**Comment by robertnishihara** [comment]
It looks like the same error also occurred here. https://travis-ci.org/robertnishihara/ray/jobs/210171415

**Comment by robertnishihara** [comment]
This was most likely addressed by #310.
